src/RigidBody/Joints.cpp

```diff
@@ -309,32 +309,74 @@ const  // a spatialVector per platform
 std::vector<RigidBodyDynamics::Math::SpatialVector> * rigidbody::Joints::combineExtForceAndSoftContact(
         std::vector<utils::SpatialVector> *f_ext,
         const rigidbody::GeneralizedCoordinates& Q,
-        const rigidbody::GeneralizedVelocity& QDot)
+        const rigidbody::GeneralizedVelocity& QDot,
+        std::vector<utils::Vector> *f_contacts,
+        bool updateKin)
 {
+#ifdef BIORBD_USE_CASADI_MATH
+    updateKin = true;
+#endif
+    if (updateKin) {
+        UpdateKinematicsCustom(&Q, nullptr, nullptr);
+    }
+    updateKin = false;
```

Comment:
Add guards to keep updateKin to true if Casadi

Reply:
done

Reply:
is it necessary to update QDot ? Because I did it for rigidContactToSpatialVector which only requires an update of Q.